subsys/net/lib/aws_iot/src/aws_iot.c

```diff
@@ -216,6 +224,32 @@ static void aws_fota_cb_handler(struct aws_fota_event *fota_evt)
 }
 #endif
 
+static int aws_iot_set_host_name(char *const host_name)
+{
+	int err;
+#if defined(CONFIG_AWS_IOT_BROKER_HOST_NAME_APP)
+	err = snprintf(aws_host_name_buf, sizeof(aws_host_name_buf),
+		       AWS_BROKER_HOST_NAME_PREFIX, host_name);
```

Comment:
This approach seems faulty, given that the assumption is that `host_name` might not be NULL-terminated (as `host_name_len` has been introduced I think this is a foreseeable scenario). In such case, snprintf will fill the buffer with rubbish until it encounters some stray NULL byte, of fills out the buffer. Given that `host_name_len` is known, I'd just use `memcpy` here + add NULL terminator at the end of the string.

Reply:
My bad, good catch